Don't crash into floor

Finn6360 opened the pull request:
Makes it not crash into floor.

msoucy commented on src/main/java/frc/robot/subsystems/Arm.java:
```diff
@@ -7,19 +7,32 @@
+    final double PIVIOT_HEIGHT = 46.654;
+    private double ARM_ANGLE;
+
+    NetworkTableInstance inst = NetworkTableInstance.create();
```
Can this be done on startup? Or does it have to wait for something else to come up? Also you're creating an entire table?

msoucy replied:
I *highly* recommend using `getDefault` here, per [the documentation](https://docs.wpilib.org/en/stable/docs/software/networktables/tables-and-topics.html)

msoucy commented on src/main/java/frc/robot/subsystems/Arm.java:
```diff
@@ -7,19 +7,32 @@
+    }
+
+    public boolean intakeBelowGround() {
+        return (Math.cos(ARM_ANGLE) * data.distanceInches) - PIVIOT_HEIGHT < 0;
```
`x - y < 0` is just `x < y` for people who want to look fancy

msoucy commented on src/main/java/frc/robot/subsystems/Arm.java:
```diff
@@ -7,19 +7,32 @@
     public final double SPEED = 0.3;
     private final double RETRACT_SPEED = -0.1;
+    final double PIVIOT_HEIGHT = 46.654;
+    private double ARM_ANGLE;
```
Capitals are for constants. And cookies are for closers.

msoucy commented on src/main/java/frc/robot/subsystems/Arm.java:
```diff
@@ -113,14 +126,19 @@ public void safeState() {
-                && speed < 0)) {
+        if ((data.distanceInches > extendMap.softMaxDistance && speed > 0)
+                || (data.distanceInches < extendMap.softMinDistance && speed < 0)) {
+            if (speed > 0 && intakeBelowGround()) {
```
This if statement could be before the previous block

msoucy commented on src/main/java/frc/robot/subsystems/ArmRotate.java:
```diff
@@ -61,11 +74,17 @@ public void periodic() {
         if ((data.degrees > this.map.topAngle && speed > 0) ||
                 (data.degrees < this.map.bottomAngle && speed < 0)) {
+
+            if (speed < 0 && intakeBelowGround()) {
```
This if should be before the enclosing if

msoucy commented on src/main/java/frc/robot/subsystems/ArmRotate.java:
```diff
@@ -32,6 +40,11 @@ public CommandBase move(DoubleSupplier rotationSpeed) {
     }
 
+    public boolean intakeBelowGround() {
+        return (Math.cos(data.degrees) * ARM_LENGTH) - PIVIOT_HEIGHT < 0;
```
`x - y < 0`

msoucy commented on src/main/java/frc/robot/subsystems/ArmRotate.java:
```diff
@@ -17,8 +20,13 @@ public class ArmRotate extends SmartSubsystemBase {
     final double MOVE_SPEED = 0.5;
     final double COMPARE_ANGLE = 5;
     final double SLOW_DOWN = 0.1;
+    final double PIVIOT_HEIGHT = 46.654;
```
This is not how "pivot" is spelled

TimB-87 committed:
Set Some Button Bindings

TimB-87 committed:
Update Solenoid IDs

TimB-87 committed:
Update CanCoder Offsets

Lucimoot committed:
Fixed color commands

changed from onTrue to WhileTrue so it defaults when button is released

TimB-87 committed:
Integration

bot190 committed:
Merge remote-tracking branch 'origin/buttons' into robotMapUpdates

Lucimoot committed:
Cube Intake Commands fixed

makes it so the motors stop once the command is done

Lucimoot committed:
Fixed an oopsie

swapped the motor stop end command from coneRelease to cubeRelease, whoops

msoucy commented:
Since the encoders get zeroed on startup, the safest thing to do would be to have the encoder get reset when the arm is fully extended or retracted.
When the current in the motor is way too high and the speed is negative, then it's probably trying to retract but it's all the way retracted. In that case, set the encoder value to 0.
When the current in the motor is way too high and the speed is positive, then it's probably trying to extend but it's all the way extended. In that case, set the encoder to `MAX_EXTENDED` value.

TimB-87 committed:
More Integration

TimB-87 committed:
Create Hard Limits

Creates hard limits besides the actual settings for them

TimB-87 commented:
I believe I did what you wanted Matt. We had the retract stopper and Ben said we probably don't need the extend stopper. I did, however, add hard limits.

TimB-87 committed:
Fix small errors

Commit:
reset encoder

TimB-87 committed:
Set reset for extension and configure button bindings

Commit:
Make sure arm don't crash into ground.


Add to not crash into ground


Fixed math for intake height and added network tables


Tims thing fixed


Tims thing


added gear ratio to get degrees


Fixed up bugs 


got rid of unnecessary code and added real units to Tims code


Fixed Matts comments


Changed create to get defult


Added nt to inst


reset encoder


build sucsesful
